File: java/242_Valid_Anagram.java
import java.util.Arrays;

// Solution 01

class Solution {
    public boolean isAnagram(String s, String t) {
        
        char[] tempArrayOne = s.toCharArray();
        char[] tempArrayTwo = t.toCharArray();
        
        Arrays.sort(tempArrayOne);
        Arrays.sort(tempArrayTwo);
        
        String sortedS = new String(tempArrayOne);
        String sortedT = new String(tempArrayTwo);
        
        if(sortedS.equals(sortedT)) {
            return true;
        }
            
        return false;
    }
}

// Solution 02

class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length() != t.length()) {
            return false;
        }
        
        if (s.equals(t)) {
            return true;
        }
        
        Map<Character, Integer> sHashMap = new HashMap<>();
        Map<Character, Integer> tHashMap = new HashMap<>();
        
        for (int i = 0; i < s.length(); i++) {
            sHashMap.merge(s.charAt(i), 1, Integer::sum);
            tHashMap.merge(t.charAt(i), 1, Integer::sum);
        }
        
        for (Character i : sHashMap.keySet()) {
            if(!sHashMap.get(i).equals(tHashMap.get(i))) {
                return false;
            }
        }
        
        return true;
    }
}

File: python/242_Valid_Anagram.py
# Solution 01

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        sorted_s = sorted(s)
        sorted_t = sorted(t)
        
        if (sorted_s == sorted_t):
            return True
        
        return False

# Solution 02

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s) == Counter(t)

# Solution 03

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
       
        if (len(s) != len(t)):
            return False
        
        count_s, count_t = {}, {}
        
        for i in range(len(s)):
            count_s[s[i]] = 1 + count_s.get(s[i], 0)
            count_t[t[i]] = 1 + count_t.get(t[i], 0)
            
        for j in count_s:
            if (count_s[j] != count_t.get(j, 0)):
                return False
            
        return True